ocaml/src/cli_asd.ml

```diff
@@ -365,6 +365,100 @@ let osd_bench_cmd =
   in
   osd_bench_t, info
 
+let asd_multistatistics long_ids to_json verbose cfg_file tls_config clear =
+  begin
+    let process_results results =
+      if to_json
+      then
+        begin
+          let result_to_json rs =
+            let x  =
+              List.map
+                (fun (long_id, statso) ->
+                  match statso with
+                  | None       -> long_id, `Null
+                  | Some stats -> long_id, Alba_json.AsdStatistics.to_yojson stats
+                ) rs
+            in
+            `Assoc x
+          in
+          print_result results result_to_json
+        end
+      else
+        let f =
+          (fun (long_id, statso) ->
+            let v = match statso with
+              | None -> "could not retrieve"
+              | Some stats ->
+                 Asd_statistics.AsdStatistics.show_inner
+                   stats
+                   Asd_protocol.Protocol.code_to_description
+            in
+            Lwt_io.printlf "%s : %s " long_id v
+          )
+      in Lwt_list.iter_s f results
+    in
+    let t () = 
+      with_alba_client
+        cfg_file tls_config
+        (fun alba_client ->
+          let open Nsm_model.OsdInfo in
+          alba_client # mgr_access # list_all_claimed_osds >>= fun (_n, osds) ->
```

Comment:
This is inefficient when the call is used with only a fraction of the long-ids, and it doesn't work for non-claimed asds.
I think it would be better to add a multi-get-osds call and use that with all the specified long ids.